lib_json/src/main/x/json/JsonBuilder.x

```diff
@@ -104,66 +102,65 @@ class JsonBuilder<JsonType extends JsonStruct, Id extends Int | String> {
-        for (Map.Entry<String, Doc> entry : o.entries) {
+    protected void mergeObject(JsonObject jsonObj) {
+        for (Map.Entry<String, Doc> entry : jsonObj.entries) {
             deepMerge(JsonPointer.from(entry.key), entry.value);
         }
     }
 
     /**
      * Deeply merge a `JsonObject` value into this builder.
      *
-     * @param p     the `JsonObject` to merge
+     * @param obj   the `JsonObject` to merge
      * @param path  the path to the location the object value should be merged into
      * @param doc   the value to merge the object into
      * @param id    the id of the entry being merged into
      */
-    protected void mergeObjectMember(JsonObject o, JsonPointer path, Doc doc, Id id) {
+    protected void mergeObjectMember(JsonObject obj, JsonPointer path, Doc doc, Id id) {
         JsonPointer remainder = path.remainder ?: assert;
-        JsonObject  updated   = new JsonObjectBuilder(o).deepMerge(remainder, doc).build();
+        JsonObject  updated   = new JsonObjectBuilder(obj).deepMerge(remainder, doc).build();
         update(id, updated);
     }
 
     /**
      * Deeply merge the entries in a `JsonArray` into the JSON value being
      * produced by this builder.
      *
-     * @param a  the `JsonArray` to merge
+     * @param array  the `JsonArray` to merge
      */
-    protected void mergeArray(JsonArray a) {
-        for (Int i : 0 ..< a.size) {
-            deepMerge(JsonPointer.from(i.toString()), a[i]);
+    protected void mergeArray(JsonArray array) {
+        for (Int i : 0 ..< array.size) {
+            deepMerge(JsonPointer.from(i.toString()), array[i]);
         }
     }
 
     /**
      * Deeply merge a `JsonArray` value into this builder.
      *
-     * @param a     the `JsonArray` to merge
-     * @param path  the path to the location the array value should be merged into
-     * @param doc   the value to merge the array into
-     * @param id    the id of the entry being merged into
+     * @param array  the `JsonArray` to merge
+     * @param path   the path to the location the array value should be merged into
+     * @param doc    the value to merge the array into
+     * @param id     the id of the entry being merged into
      */
-    protected void mergeArrayMember(JsonArray a, JsonPointer path, Doc doc, Id id) {
+    protected void mergeArrayMember(JsonArray array, JsonPointer path, Doc doc, Id id) {
         JsonPointer remainder = path.remainder ?: assert;
-        JsonArray   updated   = new JsonArrayBuilder(a).deepMerge(remainder, doc).build();
+        JsonArray   updated   = new JsonArrayBuilder(array).deepMerge(remainder, doc).build();
         update(id, updated);
     }
 
     /**
-     * Deeply merge a `Primitive` value into this builder.
+     * Deeply merge a `Primitive` json value into this builder.
      *
-     * @param p     the `Primitive` to merge
+     * @param obj   the `Primitive` value to merge (TODO JK: value is not used!?)
```

Comment:
Right, it can be removed.

Reply:
Hmm... but if we remove it, then what is the point of calling this method? Don't we lose the value?

Reply:
So you can merge one json object into another, or one json array into another, but you cannot merge anything into a primitive. So if the target field is a primitive, it is basically just replaced with the value being merged in.

e.g. If I have
```
{"foo": 1234}
```

so "foo" is a primitive number field. The I want to merge the array `[1, 2, 3, 4]` into field "foo" I end up with
```
{"foo": [1, 2, 3, 4]}
```

If "foo" was an array
```
{"foo": [9, 8, 7]}
```
 and I did the same merge then I believe I would get
```
{"foo": [9, 8, 7, 1, 2, 3, 4]}
```

Reply:
And, yes we lose the original value if it was a primitive

Reply:
Oh... I got confused by the doc then. The method doc says: 

    Deeply merge a `Primitive` value into this builder. 
    ...
    @param doc   the value to merge the primitive into
    
So I thought the current value is supposed to be replaced by the primitive. I think it needs to be improved